File: 228.summary.ranges.rb
# @param {Integer[]} nums
# @return {String[]}
def summary_ranges(nums)
    def range2str(start, finish)
        return start == finish ? (start.to_s) : ("#{start}->#{finish}")
    end
    ret = []
    start = nil
    finish = nil
    for i in nums
        if start == nil
            start = i
            finish = i
        elsif finish + 1 == i
            finish = i
        else
            ret << range2str(start, finish)
            start = i
            finish = i
        end
    end
    if start != nil
        ret << range2str(start, finish)
    end
    return ret
end

File: 228. Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        n = len(nums)
        if n == 0:
            return []
        ret = []
        start = 0
        for i in range(1, n):
            if nums[i-1] + 1!= nums[i]:
                if nums[start] == nums[i-1]:
                    ret.append(f'{nums[start]}')
                else:
                    ret.append(f'{nums[start]}->{nums[i-1]}')
                start = i
        if nums[start] == nums[-1]:
            ret.append(f'{nums[start]}')
        else:
            ret.append(f'{nums[start]}->{nums[-1]}')
        return ret